typescript/cli/src/deploy/warp.ts

```diff
@@ -143,6 +156,34 @@ async function runDeployPlanStep({ context, warpDeployConfig }: DeployParams) {
   if (!isConfirmed) throw new Error('Deployment cancelled');
 }
 
+/**
+ * Validates that the ISM configurations are compatible with each chain's technical stack.
+ * Throws an error if an incompatible ISM type is configured for a chain.
+ */
+function validateIsmCompatibility(
+  warpRouteConfig: WarpRouteDeployConfig,
+  context: WriteCommandContext,
+) {
+  for (const chain of Object.keys(warpRouteConfig)) {
+    const config = warpRouteConfig[chain];
+    const { technicalStack: chainTechnicalStack } =
+      context.multiProvider.getChainMetadata(chain);
+
+    if (
+      config.interchainSecurityModule &&
+      typeof config.interchainSecurityModule !== 'string'
+    ) {
+      assert(
+        isIsmCompatible({
+          chainTechnicalStack,
+          ismType: config.interchainSecurityModule.type,
+        }),
+        `Selected ISM of type ${config.interchainSecurityModule.type} is not compatible with the selected Chain Technical Stack of ${chainTechnicalStack} for chain ${chain}!`,
+      );
+    }
+  }
```

Comment:
Isn't this quite similar to [this](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/b6d03cc18301843dddb6d4d5561a23f57fe8c582/typescript/cli/src/deploy/core.ts#L163-L180) with the only difference that here we iterate over each chain?

Reply:
yeah exactly - will be refactoring as part of the move to modules once the SDK pr is merged

Reply:
so theyre similar but not exact - still refactored so it reuses more logic